### ISSUE TITLE
[working-1.6] Correct for schema divergence to dev-branch

### ISSUE THREAD

**Issue by Sebastian-Roth** [FOGProject/fogproject#426]
The `dmidecodeKeys` table ([see here](https://github.com/FOGProject/fogproject/blob/working-1.6/packages/web/commons/schema.php#L3786)) is missing if you update from dev-branch to working-1.6 but it does exist on a fresh working-1.6 install.

Also in the `multicastSessions` table we seem to have a difference between upgrade (dev-branch to working-1.6) and a fresh working-1.6 install. Column renames are missing on an upgrade install:
* msAnon3 -> msShutdown
* msAnon4 -> msMaxwait

**Comment by mastacontrola** [comment]
So I performed a test with dev-branch and msShutdown and msMaxwait are already defined. (This is a clean database using only dev-branch.)

I haven't attempted installing working-1.6 database yet. So the issue in the link appears to be something unrelated?

I am still looking to try to figure out about the dmedecodeKeys though.

**Comment by mastacontrola** [comment]
I have added a secondary snippet as schema version 283 using a CREATE TABLE IF NOT EXISTS for the dmidecodeKeys and this appears to have generated the proper code.

**Comment by Sebastian-Roth** [comment]
@mastacontrola Sorry but I need to re-open this issue. Thanks for adding the dmidecodeKeys! Though the differences in the `multicastSessions` table still seem to exist. Steps to reproduce:

1. Install fresh VM with Ubuntu 20.04 (I am sure it doesn't matter which OS is used but just for reference), update and take a snapshot.
2. Install FOG dev-branch (1.5.9.111 as of today) and check things are working.
3. Update to FOG working-1.6 (1.6.0-alpha.1127) and make sure all is fine.
4. Take a dump from the database: `mysqldump -u root -p --allow-keywords --result-file=dev-branch_updated_to_working-1.6.sql fog` (and copy to host system)
5. Reset the VM to the snapshot created at step one.
6. Install work FOG working-1.6 (1.6.0-alpha.1127) and make sure all is fine.
7. Take a dump from the database: `mysqldump -u root -p --allow-keywords --result-file=working-1.6.sql fog` (and copy to host system)
8. Compare the two SQL dumps: `diff -Nur working-1.6.sql dev-branch_updated_to_working-1.6.sql`

Beside unrelated differences like passwords/UUID I still see this difference in table `multicastSessions`:
```
   `msCompleteDateTime` datetime NOT NULL,
   `msIsDD` int(11) NOT NULL,
   `msNFSGroupID` int(11) NOT NULL,
-  `msShutdown` enum('0','1') NOT NULL DEFAULT '0',
-  `msMaxwait` int(11) NOT NULL,
+  `msAnon3` varchar(250) NOT NULL,
+  `msAnon4` varchar(250) NOT NULL,
   `msAnon5` varchar(250) NOT NULL,
   PRIMARY KEY (`msID`),
   KEY `new_index` (`msNFSGroupID`)
```

**Comment by mastacontrola** [comment]
Figured out what happened, it seems to have been a double spelling mistake.

Specifically looking at https://github.com/FOGProject/fogproject/blob/working-1.6/packages/web/commons/schema.php#L3975 and L3985, columnAnon3 and columnAnon4 should have been $columnmsAnon3 and $columnmsAnon4.

columnmsShutdown and columnmsMaxwait was called earlier as $columnmsmsShutdown $columnmsmsShutdown

I have switched the earlier variables to lose the double ms'. I updated the Anon variables as needed as well.

I did test it, but it failed on the first install as I think it was stuck iterating through the schema update. Second install completed without any issue. I'm going to add a loop to the schema process to have a random delay for each item. The delay would be in ms so may take a couple seconds to run the update, but should prevent issues with attempting to look at things faster than the system can process.
